sdks/python/apache_beam/ml/inference/base.py

```diff
@@ -1434,19 +1519,27 @@ def load():
     if isinstance(side_input_model_path, str) and side_input_model_path != '':
       model_tag = side_input_model_path
     if self._model_handler.share_model_across_processes():
-      model = multi_process_shared.MultiProcessShared(
-          load, tag=model_tag, always_proxy=True).acquire()
+      models = []
+      for i in range(self._model_handler.model_copies()):
+        models.append(
+            multi_process_shared.MultiProcessShared(
```

Comment:
do we need to explicitly release the multi-process shared handle for it to be freed-up? Guessing we currently don't do that as per TODO on line 1516?

If we are using the update via side input mechanism, are we reusing the same shared handle, or will side input update result in allocating more memory

Reply:
> will side input update iteration result in creating a new multiprocess shared handle without releasing the previous one 

As far as I can tell, we will  recreate a new model_wrapper after calling _RunInferenceDoFn.update_model()  from https://github.com/apache/beam/blob/58bea3f19e5220c08a900a3005e80473561dabaf/sdks/python/apache_beam/ml/inference/base.py#L1525 

and then we will create a new ModelWrapper with a new SharedHandle(s) in: https://github.com/apache/beam/blob/58bea3f19e5220c08a900a3005e80473561dabaf/sdks/python/apache_beam/ml/inference/base.py#L1474

Reply:
> do we need to explicitly release the multi-process shared handle for it to be freed-up? Guessing we currently don't do that as per TODO on line 1516?

We don't do that, I think it is fine for now though because we'd only want to unload models in batch mode when we transition to a new stage (when cleanup should happen anyways)

> If we are using the model update functonality via side input , are we reusing the same mutliprocess shared handle during the update , or will side input update iteration result in creating a new multiprocess shared handle without releasing the previous one (trying to answer this myself atm)?

We'll create a new one, but with the same tag. This will result in the current one being garbage collected